scripts/utils/printing_tools.js

```diff
@@ -0,0 +1,52 @@
+const BN = require("bn.js")
+const bnOne = new BN(1)
+const bnTwo = new BN(2)
+const bnTen = new BN(10)
+const bn256 = new BN(256)
+const bnMaxUint = bnTwo.pow(bn256).sub(bnOne)
+
+/**
+ * A generalized version of "toWei" for tokens with an arbitrary amount of digits.
+ * If the decimal representation has more digits than the maximum amount possible, then the extra digits are truncated.
+ * @param {string} amount User-friendly representation for the amount of some ERC20 token
+ * @param {integer} digits Maximum number of decimal digits of the token
+ * @return {string} decimal representation of the number of token units corresponding to the input amount
+ */
+const fromUserToMachineReadable = function (amount, digits) {
+  if (digits < 0 || digits > 256) throw Error("Invalid number of digits for ERC20 token: " + digits.toString())
+  const re = /^(\d+)(\.(\d+))?$/ // a sequence of at least one digit (0-9), followed by optionally a dot and another sequence of at least one digit
+  const match = re.exec(amount)
+  if (match == null) throw Error("Failed to parse decimal representation of " + amount)
+  const decimalString = (match[3] || "0").padEnd(digits, "0")
+  const integerPart = new BN(match[1])
+  const decimalPart = new BN(decimalString)
+  const representation = integerPart.mul(bnTen.pow(new BN(digits))).add(decimalPart)
+  if (representation.gt(bnMaxUint)) throw Error("Number larger than ERC20 token maximum amount")
+  return representation.toString()
+}
+
+/**
+ * A generalized version of "fromWei" for tokens with an arbitrary amount of digits.
+ * @param {string} amount Decimal representation of the (integer) number of token units 
+ * @param {integer} digits Maximum number of decimal digits of the token
+ * @return {string} Dot-separated decimal representation of the amount of token corresponding to the input unit amount
```

Comment:
I think we should just return a  floattype and then we can use the existing print functions to make it readable.